### ISSUE TITLE
macOS: buildkite-agent upgrade broke buildkite plugins

### ISSUE THREAD

**Issue by maleadt** [JuliaCI/sandboxed-buildkite-agent#89]
Some plugins rely on modern Bash features:

```
Running plugin coreupload post-command hook
$ /private/var/tmp/agent-tempdirs/default-honeycrisp-P2C257CXGN.0/home/.buildkite-agent/plugins/github-com-JuliaCI-coreupload-buildkite-plugin-v2/hooks/post-command
/private/var/tmp/agent-tempdirs/default-honeycrisp-P2C257CXGN.0/home/.buildkite-agent/plugins/github-com-JuliaCI-coreupload-buildkite-plugin-v2/hooks/post-command: line 3: shopt: globstar: invalid shell option name
🚨 Error: The plugin coreupload post-command hook exited with status 1
```

This because for some reason, `/bin/bash` somehow used to resolve to Homebrew's Bash, which is v5+. On recent versions of the Buildkite agent, `/bin/bash` (correctly?) refers to the system Bash, so we need to use `/usr/bin/env` to detect Bash in `/opt/homebrew` or `/usr/local/homebrew`.

**Comment by maleadt** [comment]
Fixed by https://github.com/JuliaCI/coreupload-buildkite-plugin/pull/17; the `#/bin/bash` shebang picked up macOS' ancient copy of Bash.

**Comment by maleadt** [comment]
Re-opening; also happens with cryptic:

```
$ /private/var/tmp/agent-tempdirs/default-honeycrisp-R17H3W25T9.0/home/.buildkite-agent/plugins/github-com-staticfloat-cryptic-buildkite-plugin-v2/hooks/post-checkout
/private/var/tmp/agent-tempdirs/default-honeycrisp-R17H3W25T9.0/home/.buildkite-agent/plugins/github-com-staticfloat-cryptic-buildkite-plugin-v2/lib/common.sh: line 5: shopt: globstar: invalid shell option name
🚨 Error: The plugin cryptic post-checkout hook exited with status 1
```

This breaks the upload task of the agents: https://buildkite.com/julialang/julia-master/builds/41488#0192bf05-caf1-46b5-8eca-698816f765b9

Should be fixed by https://github.com/staticfloat/cryptic-buildkite-plugin/pull/31.

**Comment by maleadt** [comment]
`cryptic` has been fixed, and v2.2 makes the upload steps work again: https://buildkite.com/julialang/julia-master/builds/41488#0192c24b-dc75-4c5f-86c8-f8304dab27b2
